File: 2849-determine-if-a-cell-is-reachable-at-a-given-time/2849-determine-if-a-cell-is-reachable-at-a-given-time.java
class Solution {
    public boolean isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int dx = Math.abs(fx-sx);
        int dy = Math.abs(fy-sy);
        if(dx==0 && dy==0)
            return t!=1;
        int mn = Math.min(dx, dy);
        int straight = Math.max(dx, dy) - mn;
        return t>=mn+straight;
    }
}

File: 2849-determine-if-a-cell-is-reachable-at-a-given-time/2849-determine-if-a-cell-is-reachable-at-a-given-time.py
class Solution:
    def isReachableAtTime(self, sx: int, sy: int, fx: int, fy: int, t: int) -> bool:
        dx, dy = abs(fx-sx), abs(fy-sy)
        if dx==0 and dy==0:
            return t!=1
        return t >= max(dx, dy)